attach.c

```diff
@@ -115,6 +123,21 @@ int do_setsid(struct ptrace_child *child) {
     return err;
 }
 
+static int do_dup2(struct ptrace_child *child, int oldfd, int newfd) {
+    if (ptrace_syscall_numbers(child)->nr_dup2 != -1) {
+        return do_syscall(child, dup2, oldfd, newfd, 0, 0, 0, 0);
+    } else {
+        int err = do_syscall(child, dup3, oldfd, newfd, 0, 0, 0, 0);
+        // dup3 returns EINVAL when the fds are the same, while dup2 ignores it
```

Comment:
`dup3` returns `EINVAL` for other reasons, too. They're probably not relevant here, but let's instead check for the equality of the fds and shortcut the call.

Reply:
(applied a fix in #98)